flocker/apiclient/_client.py

```diff
@@ -41,16 +41,18 @@ class DatasetState(PClass):
     """
     The state of a dataset in the cluster.
 
-    :attr UUID primary: The node where the dataset should manifest.
-    :attr int maximum_size: Size of new dataset, in bytes.
+    :attr primary: The ``UUID`` of the node where the dataset is manifest,
+        or ``None`` if has no manifestation.
+    :attr maximum_size: Size of new dataset in bytes, or ``None`` if it
+        has none.
```

Comment:
Again, remove the "new".

This definition implies that `maximum_size` really means actual size, in which case, what does it mean for a dataset state to have no actual size (`maximum_size is None`)?  Alternatively, if `maximum_size` is a limit and it is guaranteed that the actual size <= `maximum_size`, then `None` makes sense, but `maximum_size` is not the "size of the dataset".  Please clarify the definition.